extension/beatsauthextension/authenticator.go

```diff
@@ -35,32 +37,33 @@ var _ extensionauth.GRPCClient = (*authenticator)(nil)
 var _ extension.Extension = (*authenticator)(nil)
 
 type authenticator struct {
-	cfg       *Config
-	telemetry component.TelemetrySettings
-	tlsConfig *tlscommon.TLSConfig // set by Start
-	logger    *logp.Logger
+	telemetry    component.TelemetrySettings
+	httpSettings httpcommon.HTTPTransportSettings
+	logger       *logp.Logger
+	client       *http.Client
 }
 
 func newAuthenticator(cfg *Config, telemetry component.TelemetrySettings) (*authenticator, error) {
 	logger, err := logp.NewZapLogger(telemetry.Logger)
 	if err != nil {
 		return nil, err
 	}
-	return &authenticator{cfg: cfg, telemetry: telemetry, logger: logger}, nil
+
+	parsedCfg, err := config.NewConfigFrom(cfg.BeatAuthconfig)
```

Comment:
This function is what will load certificates and CAs from disk, where do those failures end up being discovered when this extension is part of a running collector that is starting up?